### PR TITLE
Fixed build failed because of eslint errors  - Also fixed small typos in several places

### PR DESCRIPTION
**Pull request by chrischang**
Hey there,

Came across the repo from d3 website and wanted to try it out, but I ran into eslint errors after running `npm install && npm start`, so I decided to fix them. The graphs still work after the fix. Feel free to fork  it and test it out.

### CODE REVIEW

File: static/javascript/components/bar-charts/axis.js [hunk 1]
```diff
@@ -1,9 +1,7 @@
 import React from "react";
 import * as d3 from "d3";
 
-export default class HorisontalBarChart2 extends React.Component {
-  static displayName = "Axis";
-
+export default class HorizontalBarChart2 extends React.Component {
```
Comment (artyomtrityak) [inline static/javascript/components/bar-charts/axis.js]: why did you remove `static displayName = "Axis";` ?

File: static/javascript/components/bar-charts/group-vertical-bar-chart.js [hunk 2]
```diff
@@ -19,18 +18,16 @@ const DATA = [
 ];
 
 class SvgVerticalGroupedBarChart extends React.Component {
```
Comment (artyomtrityak) [inline static/javascript/components/bar-charts/group-vertical-bar-chart.js]: object level properties works fine, is there a reason you removed them?

**Review by artyomtrityak — COMMENTED** [review]
class fields are fine https://github.com/tc39/proposal-class-fields. Is there a reason you removed them?

**Comment by artyomtrityak** [comment]
@chrischang thanks for contributing. Class level fields you have removed are valid (Stage 2) JavaScript definitions. https://github.com/tc39/proposal-class-fields
What version of eslint are you using? Did you try `npm test` in project after `npm install`? For me it works fine.

File: static/javascript/components/bar-charts/axis.js (hunk 1 above)
Comment (chrischang) [reply static/javascript/components/bar-charts/axis.js]: I got an error when running it  15:10  error  'displayName' is not defined  no-undef
"eslint": "^3.15.0"

I bumped the version up to ^4.15.0, and it worked fine.

I did a search and it seems like it's a known issue with eslint:
https://github.com/eslint/eslint/issues/8720

I did a search for displayName in the files, and did not find anywhere that uses `displayName`, so I reckon it was safe to remove.

File: static/javascript/components/bar-charts/group-vertical-bar-chart.js [hunk 3]
```diff
@@ -1,6 +1,5 @@
 import React from "react";
 import * as d3 from "d3";
-import { generateArray } from "../../data-layer/array-processors";
```
Comment (chrischang) [inline static/javascript/components/bar-charts/group-vertical-bar-chart.js:3]: This isn't used.

File: static/javascript/components/bar-charts/group-vertical-bar-chart.js (hunk 2 above)
Comment (chrischang) [reply static/javascript/components/bar-charts/group-vertical-bar-chart.js]: there was an eslint error when I did `npm start`

File: static/javascript/components/bar-charts/axis.js (hunk 1 above)
Comment (artyomtrityak) [reply static/javascript/components/bar-charts/axis.js]: displayName is used in React itself. https://reactjs.org/docs/react-component.html

Comment (chrischang) [reply static/javascript/components/bar-charts/axis.js]: Ah, alright. Thanks for the doc. I could add them back.

**Comment by chrischang** [comment]
Updated! Let me know if you have any other feedback.

**Comment by artyomtrityak** [comment]
You do not need to change class level fields. They are valid and work correct
